client/src/main/java/org/apache/falcon/client/FalconClient.java

```diff
@@ -1073,16 +1073,16 @@ private FormDataMultiPart getEntitiesForm(String extensionName, String jobName,
         List<Entity> entities = validateExtensionAndGetEntities(extensionName, jobName, configStream);
         FormDataMultiPart formDataMultiPart = new FormDataMultiPart();
 
-        for (Entity entity : entities) {
-            if (EntityType.FEED.equals(entity.getEntityType())) {
-                formDataMultiPart.field("feeds", entity, MediaType.APPLICATION_XML_TYPE);
-            } else if (EntityType.PROCESS.equals(entity.getEntityType())) {
-                formDataMultiPart.field("processes", entity, MediaType.APPLICATION_XML_TYPE);
+        if (entities != null && !entities.isEmpty()) {
+            for (Entity entity : entities) {
+                if (EntityType.FEED.equals(entity.getEntityType())) {
+                    formDataMultiPart.field(FEEDS, entity, MediaType.APPLICATION_XML_TYPE);
```

Comment:
Have you tested this with multiple feeds and processes? From the looks of it, you are overwriting the same fieldName with different entity definition.